File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    n = len(elements)
    result = set()
    
    extended_elements = elements * 2
    
    for length in range(1, n + 1):
        for start in range(n):
            s = sum(extended_elements[start : start + length])
            result.add(s)
            
    return len(result)

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    const n = elements.length;
    const extended = [...elements, ...elements];
    const results = new Set();

    for (let len = 1; len <= n; len++) {
        for (let start = 0; start < n; start++) {
            const sum = extended.slice(start, start + len).reduce((acc, cur) => acc + cur, 0);
            results.add(sum);
        }
    }

    return results.size;
}